plugins/modules/controller.py

```diff
@@ -0,0 +1,214 @@
+    - name: Create device within Nautobot with empty string name to generate UUID
+      networktocode.nautobot.device:
+        url: http://nautobot.local
+        token: thisIsMyToken
+        name: ""
+        device_type: C9410R
+        role: Core Switch
+        location:
+          name: My Location
+          parent: Parent Location
+        status: active
+        state: present
+
+    - name: Delete device within nautobot
+      networktocode.nautobot.device:
+        url: http://nautobot.local
+        token: thisIsMyToken
+        name: Test Device
+        state: absent
+
+    - name: Create device with tags
+      networktocode.nautobot.device:
+        url: http://nautobot.local
+        token: thisIsMyToken
+        name: Another Test Device
+        device_type: C9410R
+        role: Core Switch
+        location:
+          name: My Location
+          parent: Parent Location
+        status: active
+        local_config_context_data:
+          bgp: "65000"
+        tags:
+          - Schnozzberry
+        state: present
+
+    - name: Update the rack and position of an existing device
+      networktocode.nautobot.device:
+        url: http://nautobot.local
+        token: thisIsMyToken
+        name: Test Device
+        rack: Test Rack
+        position: 10
+        face: Front
+        state: present
+"""
+
+RETURN = r"""
+device:
+  description: Serialized object as created or already existent within Nautobot
+  returned: success (when I(state=present))
+  type: dict
+msg:
+  description: Message indicating failure or info about what has been achieved
+  returned: always
+  type: str
+"""
+
+from ansible_collections.networktocode.nautobot.plugins.module_utils.utils import NAUTOBOT_ARG_SPEC
+from ansible_collections.networktocode.nautobot.plugins.module_utils.dcim import (
+    NautobotDcimModule,
+    NB_CONTROLLERS,
+)
+from ansible.module_utils.basic import AnsibleModule
+from copy import deepcopy
+import uuid
+
+
+def main():
+    """
+    Main entry point for module execution
+    """
+    argument_spec = deepcopy(NAUTOBOT_ARG_SPEC)
+    argument_spec.update(
+        dict(
+            name=dict(required=True, type="str"),
+            controller_device=dict(required=False, type="str"),
+            external_integration=dict(required=False, type="str"),
+            role=dict(required=False, type="raw"),
+            tenant=dict(required=False, type="raw"),
+            platform=dict(required=False, type="raw"),
+            location=dict(required=False, type="raw"),
+            status=dict(required=False, type="raw"),
+            tags=dict(required=False, type="list", elements="raw"),
+            custom_fields=dict(required=False, type="dict"),
+            controller_device_redundancy_group=dict(required=False, type="raw"),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    if module.params["name"] == "":
+        module.params["name"] = str(uuid.uuid4())
```

Comment:
Name is a required field for both the UI form and the API. I don't think we should be automatically populating this if they provide an empty string.